api/envoy/extensions/filters/http/json_to_metadata/v3/json_to_metadata.proto

```diff
@@ -108,6 +110,9 @@ message JsonToMetadata {
     // Allowed empty content-type for json to metadata transformation.
     // Default to false.
     bool allow_empty_content_type = 3;
+
+    // Allow content-type by regex match
```

Comment:
```suggestion
    // Allowed content-type by regex match
```

Reply:
Also suggest to have explaination on what if we have both allow_content_types and allow_content_types_regex

Reply:
also have some unit tests on both fields are present, where the content type is (a) in the allow_content_types (b) in the regex and (c) neither

Reply:
> also have some unit tests on both fields are present, where the content type is (a) in the allow_content_types (b) in the regex and (c) neither

add more unit tests, but (c) neither won't work at this time, probably you can add it after your fix

Reply:
for (c) i was thinking proto is 
```

  allow_content_types:
  - text/plain
  allow_content_types_regex:
  - google_re2: {}
    regex: "application/.*"
```

and the mock content type is `image/png` so probably nothing is triggered at the moment. and i'll do the follow up.